### PR TITLE
{compiler}[dummy] GCC 5.3.0 (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86: please review

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/9602dd83fcd70f4f685b for a full test report.

File: easybuild/easyconfigs/g/GCC/GCC-5.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    'ftp://gcc.gnu.org/pub/gcc/infrastructure/',  # GCC dependencies
+    'http://gcc.cybermirror.org/infrastructure/',  # HTTP mirror for GCC dependencies
+]
+
+sources = [
+    SOURCELOWER_TAR_BZ2,
+    'gmp-6.1.0.tar.bz2',
+    'mpfr-%s.tar.gz' % mpfr_version,
+    'mpc-1.0.3.tar.gz',
+    'isl-0.15.tar.bz2',
```
Comment (wpoely86) [inline easybuild/easyconfigs/g/GCC/GCC-5.3.0.eb]: don't bump isl UNLESS it's in the docs.

Comment (boegel) [inline easybuild/easyconfigs/g/GCC/GCC-5.3.0.eb]: which docs?

Comment (wpoely86) [inline easybuild/easyconfigs/g/GCC/GCC-5.3.0.eb]: https://gcc.gnu.org/gcc-5/changes.html third line

Comment (wpoely86) [inline easybuild/easyconfigs/g/GCC/GCC-5.3.0.eb]: a, it's been added on https://gcc.gnu.org/install/prerequisites.html

keep it

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/1b0fdb86d29ce5b8ebc1 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/3409523443cee0d8c528 for a full test report.

**Comment by wpoely86** [comment]
Please hold the merge until ML discussion is finished

**Comment by boegel** [comment]
This is good to go, there's no need to ensure ABI compatibility with GCC 4.x since they're not used together in the `foss/2016a` toolchain (which only uses GCC 4.9.3)

@ocaisa: I you have a working GCC v5.3.0 on top of GCCCore 4.9.3, I'd love to see a PR for it!

**Comment by ocaisa** [comment]
There is a subtle issue with GCC on top of GCCCore, if you list GCCCore as a dep of GCC then the libs of GCCCore take precedence when you use GCC as a toolchain because of how the `LD_LIBRARY_PATH` is set up. The solution is to set GCCCore as a build dep but this leads to a problem in a hierarchical scheme as you will not get the modulepath extension to access the binutils built with GCCCore. @geimer I think the solution would be to handle this special case in the naming scheme: if GCCCore is a build dep of GCC then add the GCCCore path extension to the module header (i.e., extend the path but don't load the module)

I'm happy to implement next week.

**Comment by boegel** [comment]
@ocaisa: should we open a dedicated issue for this?

**Comment by ocaisa** [comment]
Done #2371

**Pull request merged**